seed/api/v2_1/views.py

```diff
@@ -122,3 +125,82 @@ def building_sync(self, request, pk):
         """
         return JsonResponse(
             {"status": "error", "message": "Not yet implemented. PK was {}".format(pk)})
+
+    @detail_route(methods=['PUT'])
+    def update_with_building_sync(self, request, pk):
+        """
+        Does not work in Swagger!
+
+        Update an existing PropertyView with a building file. Currently only supports BuildingSync.
+        ---
+        consumes:
+            - multipart/form-data
+        parameters:
+            - name: pk
+              description: The PropertyView to update with this buildingsync file
+              type: path
+              required: true
+            - name: organization_id
+              type: integer
+              required: true
+            - name: cycle_id
+              type: integer
+              required: true
+            - name: file_type
+              type: string
+              enum: ["Unknown", "BuildingSync", "GeoJSON"]
+              required: true
+            - name: file
+              description: In-memory file object
+              required: true
+              type: file
+        """
+        if len(request.FILES) == 0:
+            return JsonResponse({
+                'success': False,
+                'message': "Must pass file in as a Multipart/Form post"
+            })
+
+        the_file = request.data['file']
+        file_type = BuildingFile.str_to_file_type(request.data.get('file_type', 'Unknown'))
+        organization_id = request.data['organization_id']
+        cycle_id = request.data.get('cycle_id', None)
+
+        if not cycle_id:
+            return JsonResponse({
+                'success': False,
+                'message': "Cycle ID is not defined"
+            })
+        else:
+            cycle = Cycle.objects.get(pk=cycle_id)
+
+        building_file = BuildingFile.objects.create(
+            file=the_file,
+            filename=the_file.name,
+            file_type=file_type,
+        )
+
+        try:
+            property_view = PropertyView.objects.get(pk=pk, cycle=cycle)
```

Comment:
I couldn't find anywhere that propertyviews were downselected to organization.  I could add it here, but it looks like maybe org is actually on the PropertyState?  View only joins the state, property, and cycle together, so I just wanted some confirmation about adding them.

Reply:
org is on property state, yes.